src/background.js

```diff
@@ -57,6 +57,12 @@ chrome.tabs.onActivated.addListener(({ tabId }) => {
 chrome.tabs.onUpdated.addListener((tabId) => {
   checkTab(tabId);
 });
+chrome.tabs.onRemoved.addListener((tabId) => {
+  checkTab(tabId);
```

Comment:
Since it about to be closed, I don't think we need to check if the tab is allowed to use this extension.

Apart from that, it works like a charm!